Source/JavaScriptCore/dfg/DFGClobberize.h

```diff
@@ -1975,17 +1976,23 @@ void clobberize(Graph& graph, Node* node, const ReadFunctor& read, const WriteFu
     case WeakSetAdd: {
         Edge& mapEdge = node->child1();
         Edge& keyEdge = node->child2();
-        write(JSWeakSetFields);
-        def(HeapLocation(WeakMapGetLoc, JSWeakSetFields, mapEdge, keyEdge), LazyNode(keyEdge.node()));
+        if (keyEdge.useKind() == ObjectUse) {
+            write(JSWeakSetFields);
+            def(HeapLocation(WeakMapGetLoc, JSWeakSetFields, mapEdge, keyEdge), LazyNode(keyEdge.node()));
+        } else
```

Comment:
This one makes sense to me, since we might throw if it's a registered symbol. But I wonder if we can still CSE it? Also, maybe we can add a speculation like NonRegisterdSymbol at some point.